app/src/main/kotlin/at/bitfire/davdroid/sync/TaskSyncer.kt

```diff
@@ -38,19 +38,17 @@ class TaskSyncer @AssistedInject constructor(
         fun create(account: Account, authority: String, extras: Array<String>, syncResult: SyncResult): TaskSyncer
     }
 
-    private lateinit var taskProvider: TaskProvider
+    private val providerName = TaskProvider.ProviderName.fromAuthority(authority)
 
     override val serviceType: String
         get() = Service.TYPE_CALDAV
 
-
     override fun localSyncCollections(provider: ContentProviderClient): List<LocalTaskList>
-        = DmfsTaskList.find(account, taskProvider, LocalTaskList.Factory, "${TaskLists.SYNC_ENABLED}!=0", null)
+        = DmfsTaskList.find(account, LocalTaskList.Factory, provider, providerName, "${TaskLists.SYNC_ENABLED}!=0", null)
 
     override fun prepare(provider: ContentProviderClient): Boolean {
-        // Acquire task provider
-        val providerName = TaskProvider.ProviderName.fromAuthority(authority)
-        taskProvider = try {
+        // Don't sync if task provider is too old
+        try {
             TaskProvider.fromProviderClient(context, providerName, provider)
         } catch (e: TaskProvider.ProviderTooOldException) {
             tasksAppManager.get().notifyProviderTooOld(e)
```

Comment:
Acquiring the provider client is responsibility of `Syncer`, so I'd move the catch for `ProviderTooOldException` there, even if it's task-specific.

(Because now we acquire the client two times. And if we would leave it here, we would still need to `close` it.)

Reply:
Hmm, not sure I like it, but I can see your point. About closing it: I just realised jtxSyncer simply calls `checkVersion`, which does not actually acquire the provider, so we can use that instead.

Reply:
Hm if we _can_ keep everything in the specific `prepare`, I think that's better. So calling `checkVersion` and catch the exception both in `prepare` would be the best thing.

I just meant that it's better to catch a specific exception in the general Syncer than acquiring the content provider a second time in `prepare` just to check the version.

Reply:
Right!